### ISSUE TITLE
Setting a message when cordoning nodes

### ISSUE THREAD

**Issue by nleib** [kubernetes/kubernetes#30586]
Hey,
when draining a node or cordon a node I think its important to be able to set a message why this node was cordoned.

e.g (two examples)

kubectl cordon K8S_NODENAME "disabled due to docker maintenance"
kubectl drain K8S_NODENAME --set-message "disabled due to docker maintenance"

**Comment by AdoHe** [comment]
@kubernetes/kubectl

**Comment by zreigz** [comment]
I would like take this task. Is it fine for you?

**Comment by nleib** [comment]
fine by me if you are referring the question to me.

**Comment by zreigz** [comment]
I think annotation is good place where I can store the message. Any other idea ?

**Comment by kargakis** [comment]
Could also be a NodeCondition.

**Comment by zreigz** [comment]
Yes it is good idea. I could set  `NodeConditionType` for new type `SchedulingDisabled` and then set the message.

**Comment by kargakis** [comment]
@smarterclayton @davidopp I assume admins should be allowed to set specific NodeConditions.

**Comment by zreigz** [comment]
For implementation point of view it should be enough to update condition Ready to state `False` with reason `SchedulingDisabled` and with given message.

```
nodeutil.SetNodeCondition(rc.kubeClient, nodeName, api.NodeCondition{
				Type:               api.Ready,
				Status:             api.ConditionFalse,
				Reason:             "SchedulingDisabled",
				Message:            "disabled due to docker maintenance",
                LastTransitionTime:   currentTime,
```

**Comment by davidopp** [comment]
cordon and drain don't make the node not-ready so I don't think that will work

I like the general idea but I'm not sure about the solution. Maybe kubectl could publish an event? Downside is it would be transient. 

cc/ @mml who might have a good idea here

**Comment by kargakis** [comment]
> cordon and drain don't make the node not-ready so I don't think that will work

I was thinking of a new condition with type="Cordoned" or something and the message would include why the node is cordoned.

**Comment by zreigz** [comment]
The events are generated for cordon/uncordon
```
Events:
  FirstSeen	LastSeen	Count	From					SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----					-------------	--------	------			-------
  37s		37s		1	{kubelet lukasz-celsius-w530-power}			Normal		NodeNotSchedulable	Node lukasz-celsius-w530-power status is now: NodeNotSchedulable
  17s		17s		1	{kubelet lukasz-celsius-w530-power}			Normal		NodeSchedulable		Node lukasz-celsius-w530-power status is now: NodeSchedulable

```

**Comment by zreigz** [comment]
Form me new condition type seems to be reasonable

**Comment by zreigz** [comment]
I've made necessary changes. You can review my PR https://github.com/kubernetes/kubernetes/pull/35113

**Comment by mml** [comment]
Assigning to myself just to mark that this issue is not 'unassigned'

**Comment by spiffxp** [comment]
/sig cli
because the changes in the linked PR appear to be in kubectl

**Comment by pacoxu** [comment]
/assign
I will take a look if I have time later.
Feel free to assign if you are interested.

**Comment by atiratree** [comment]
This feature is included in the Declarative Node Maintenance KEP: https://github.com/kubernetes/enhancements/pull/4213
